R/dvhydrograph-render.R

```diff
@@ -3,199 +3,291 @@ dvhydrographPlot <- function(data) {
 
-    plotDates <- toStartOfDay(seq(startDate, endDate, by = 7 * 24 * 60 * 60))
+  #Legend
+  legend_items <- plot_object$legend$legend.auto$legend
+  ncol <- ifelse(length(legend_items) > 3, 2, 1)
+  leg_lines <- ifelse(ncol==2, ceiling((length(legend_items) - 6)/2), 0) 
+  legend_offset <- ifelse(ncol==2, 0.3+(0.05*leg_lines), 0.3)
+  plot_object <- legend(plot_object, location="below", cex=0.8, legend_offset=0.2, y.intersp=1.5, ncol=ncol)
 
-    plot_object <- gsplot(ylog = refInfo$logAxis, yaxs = 'i') %>%
-      grid(nx = NA, ny = NULL, lty = 3, col = "gray") %>%
-      axis(2, reverse = isInverted) %>%
-      view(xlim = c(startDate, endDate)) %>%
-      title(
-        main = paste(ref_name_capital, "Reference Time Series"),
-        ylab = paste(data[[series]]$type, data[[series]]$units),
-        line = 3
-      ) %>%
-      legend(location = "below", cex = 0.8, y.intersp = 1.5)
+  #Add Min/Max labels if we aren't plotting min and max
+  minmax_labels <- append(dvData['max_iv_label'], dvData['min_iv_label'])
+  line <- 0.33
+  for(ml in na.omit(names(minmax_labels))){
+    #Extract Timezone
+    tzf <- format(as.POSIXct(dvData[[ml]]$time), "%z")
+    #Insert ":" before 2nd to last character
+    tzf <- sub("([[:digit:]]{2,2})$", ":\\1", tzf)
+    formatted_label <- paste0(dvData[[ml]]$legend.name, data$firstDownChain$units, 
+                              format(as.POSIXct(dvData[[ml]]$time), " %b %d, %Y %H:%M:%S"), " (UTC ", tzf, ")")
     
-    plot_object <-
-      XAxisLabelStyle(plot_object, startDate, endDate, data$reportMetadata$timezone, plotDates)
+    plot_object <- mtext(plot_object, formatted_label, side = 3, axes=FALSE, cex=0.85, line = line, adj = 0)
     
-    # for non-approval-bar objects
-    for (i in grep("^appr_", names(refData), invert = TRUE)) {
-      refConfig <- getDVHydrographRefPlotConfig(refData[i])
-      for (j in seq_len(length(refConfig))) {
-        plot_object <- do.call(names(refConfig[j]), append(list(object = plot_object), refConfig[[j]]))
-      }
+    line <- line + 1
+  }
+  
+  return(plot_object)
+}
+
+#' Create DV Hdyrograph Reference Plot
+#'
+#' @description Given the full report object, a series field name, and
+#' a series description field name, creates a DV Hydrograph Reference
+#' plot using the specific parameters.
+#' @param reportObject the full report JSON object
+#' @param series the series field name to extract from the JSON
+#' @param description the description field name to extract from the JSON
+createDVHydrographRefPlot <- function(reportObject, series, descriptions) {
+  #Rendering Options
+  options(scipen=8)
+
+  #Get Necessary Report Metadata
+  ref_name_capital <- switch(series,
+    'secondaryReferenceTimeSeries' = 'Secondary',
+    'tertiaryReferenceTimeSeries' = 'Tertiary',
+    'quaternaryReferenceTimeSeries' = 'Quaternary'
+  )
+  seriesEst <- paste0(series, 'Est', setp="")
+  seriesEstEdges <- paste0(series, 'EstEdges', setp="")
```

Comment:
Whoops, those are both typos of `sep` and leftover before I learned about paste0 haha.